### ISSUE TITLE
Allow ways to disable any form of data modification

### ISSUE THREAD

**Issue by andersonwinkler** [gbook/nidb#114]
If a user has "View only" access to a given project, ensure that they can never make any modification that can lead to data being entered into on modified from the backend MySQL/MariaDB.

I could also be a "hard core" read-only mode, in which a user can never do anything that could lead to any information appearing in the backend other than export logs or repeat last search (for example, disabling import, and disabling ID mapping, unless ID mapping is used only for a particular search but not for permanent storage).

In other words, a "View only" user shall not be able to move subjects around, group them, reassign IDs, or sessions, or anything. They should only ever be able to view things, and export, nothing else.
